File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>

void coins_needed(float f);

int main(void)
{
    printf ("O hai! How much change is owed? ");
    float change = get_float();
    while (change < 0){
        printf("Number must be positive. Please try again: ");
        change = get_float();
    }
    printf ("%.2f\n", change);
    coins_needed(change);
}

void coins_needed(float f)
{
    int count = 0;
    while (f > 0.0)
    {
        if (f >= .25)
        {
            f -= .25;
            count +=1;
        }
        else if (f >=.10)
        {
            f -= .10;
            count +=1;
        }
        else if (f >=.05)
        {
            f -= .05;
            count +=1;
        }
        else
        {
            f -= .01;
            count +=1;
        }
    }
    printf("%i coins needed\n", count);
}

File: pset6/greedy.py
import cs50

while True:
   change = cs50.get_float()
   if change > 0.0:
       break
   else:
       print("Retry: ")

count = 0
while change > 0.0:
    if change >= .25:
        change -=.25
        count +=1
    elif change >=.10:
        change -=.10
        count +=1
    elif change >=.05:
        change -=.05
        count +=1
    else:
        change -=.01
        count +=1
        
print ("{}".format(count))
    